src/Instrumentation/PDO/src/PDOAttributeTracker.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\PDO;
+
+use OpenTelemetry\SemConv\TraceAttributes;
+
+final class PDOAttributeTracker
+{
+    private $pdoToAttributesMap;
+    private $statementMapToPdoMap;
+
+    public function __construct()
+    {
+        $this->pdoToAttributesMap = [];
+        $this->statementMapToPdoMap = [];
+    }
+
+    public function trackStatementToPdoMapping(\PDOStatement $statement, \PDO $pdo)
+    {
+        $this->statementMapToPdoMap[spl_object_id($statement)] = spl_object_id($pdo);
+    }
+
+    public function removeMapping(\PDOStatement $statement)
+    {
+        $key = spl_object_id($statement);
+        if (array_key_exists($key, $this->statementMapToPdoMap)) {
+            unset($this->statementMapToPdoMap[$key]);
+        }
```

Comment:
```suggestion
        unset($this->statementMapToPdoMap[spl_object_id($statement)]);
```

Should work without any errors